DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1659,21 +1659,16 @@ internal async Task OnVoiceStateUpdateEventAsync(JObject raw)
             var uid = (ulong)raw["user_id"];
             var gld = this._guilds[gid];
 
-            var vstateHasNew = gld._voiceStates.TryGetValue(uid, out var vstateNew);
-            DiscordVoiceState vstateOld;
-            if (vstateHasNew)
-            {
-                vstateOld = new DiscordVoiceState(vstateNew);
-                DiscordJson.PopulateObject(raw, vstateNew);
-            }
-            else
+            var vstateNew = raw.ToObject<DiscordVoiceState>();
+            vstateNew.Discord = this;
+
+            if (!gld._voiceStates.TryGetValue(uid, out var vstateOld))
             {
                 vstateOld = null;
```

Comment:
This code is redundant since `vstateOld` will be null if it can't be found. You can probably just have `TryGetValue` here without a conditional to assign the value if present.